metrics/web/docs/features/public_dashboard/01_public_dashboard.md

```diff
@@ -137,3 +161,208 @@ Once we've chosen the implementation approach, let's consider the components dia
+- Manage normal sign in for the anonymous user.
+- Change the UI of the Metrics user menu.
+
+Let's review each of the above points in the following subsections
+
+##### Auto sign in the user anonymously
+
+To auto sign in the user anonymously, we should make the following:
+1. The `handleAuthUpdates()` method of the `NavigationNotifier` should be called when the `AuthNotifier.authState` changes.
+2. The `handlePublicDashboardFeatureConfigUpdates()` method of the `AuthNotifier` should be called when the `FeatureConfigNotifier.publicDashboardFeatureConfigModel` changes.
+3. Add the logic of the signing the user in anonymously depending on the `AuthNotifier.isPublicDashboardFeatureEnabled` and the `AuthNotifier.authState` to the `AuthNotifier`.
+4. Add the logic of the navigating depending on the `AuthNotifier.authState` to the `NavigationNotifier`.
+
+_**Note:** The `handleAuthUpdates()` does nothing if the `AuthNotifier.authState` equals to the `AuthState.loggedIn`, because we handle this case in the `LoginPage` class._
+
+##### Manage sign in for the anonymous user
+
+The next one, to manage the sign-in for the anonymous user (f.e. sign in with credentials or using Google provider), we have to sign in them using one of the sign methods mentioned above, then the Firebase auth, for its part, replaces the current user with a new one without any problems.
+
+##### Change the Metrics user menu UI
+
+The last thing we should change the Metrics user menu UI accordingly to the [User Interface section](#user-interface). For this purpose, we should take the [`UserProfileViewModel`](#UserProfileViewModel) from the `AuthNotifier` and disable the specified parts in the UI based on the view model value. 
+
+Consider the overall class diagram for the public dashboard feature that describes the relationships between the classes this feature requires:
+
+![Public Dashboard Feature Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/public_dashboard_design/metrics/web/docs/features/public_dashboard/diagrams/public_dashboard_class_diagram.puml)
+
+Consider the overall sequence diagram for the public dashboard feature that describes the main flow for the public dashboard feature:
+
+![Public Dashboard Feature Sequence Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/public_dashboard_design/metrics/web/docs/features/public_dashboard/diagrams/public_dashboard_sequence_diagram.puml)
+
+### Testing
+> Describe an approach on how the feature should be tested.
+
+To test the feature, we should focus on the following types of tests:
+- The unit tests, which cover the main logic of the feature, described in the [program section](#program).
+- The widget tests, which test that the UI behaviour corresponds to the expectations from the [User Interface section](#user-interface).
```

Comment:
🔨
I believe we should also add integration tests to test that everything works correctly together.